dotnet/src/SemanticKernel.Abstractions/Kernel.cs

```diff
@@ -54,6 +58,11 @@ public sealed class Kernel
 
         // Store the provided plugins. If there weren't any, look in DI to see if there's a plugin collection.
         this._plugins = plugins ?? this.Services.GetService<KernelPluginCollection>();
+
+        // Store the provided function and prompt filters if they were registered.
+        this._functionFilters = this.Services.GetServices<IFunctionFilter>().ToList();
+        this._promptFilters = this.Services.GetServices<IPromptFilter>().ToList();
+
```

Comment:
Do you mean that `_functionFilters` and `_promptFilters` fields should be non-nullable in this case, or you mean that we need to store these fields as `IEnumerable<T>` instead of `IList<T>` and perform `ToList()` only when we need them? (e.g. access `Count` in `Clone` method).

Reply:
I expect the most common case to optimize for here is no filters registered; if you use `ToArray` instead of `ToList`, it'll special-case empty if it can determine it's empty and return a singleton array in that case. ToList can't do that because it returns a mutable list.